zds/forum/commons.py

```diff
@@ -64,13 +68,16 @@ def perform_move(request, topic):
 
             # If the topic is moved in a restricted forum, users that cannot read this topic any more un-follow it.
             # This avoids unreachable notifications.
-            followers = TopicFollowed.objects.filter(topic=topic)
-            for follower in followers:
-                if not forum.can_read(follower.user):
-                    follower.delete()
+
+            subscriptions = TopicAnswerSubscription.objects.get_subscriptions(topic)
+            for subscription in subscriptions:
+                if not forum.can_read(subscription.profile.user):
```

Comment:
J'ai la vague impression que l'ensemble "condition + desactivate + mark as read" est un bloc unitaire qui peut être réutilisable à plein d'endroit. Je ne serais pas contre un "extract method => move to utils".